src/Sinch/Core/Http.cs

```diff
@@ -154,13 +161,27 @@ internal class Http : IHttp
                 if (result.IsJson())
                     return await result.Content.ReadFromJsonAsync<TResponse>(cancellationToken: cancellationToken,
                                options: _jsonSerializerOptions)
-                           ?? throw new NullReferenceException(
-                               $"{nameof(TResponse)} is null");
+                           ?? throw new InvalidOperationException(
+                               $"{typeof(TResponse).Name} is null");
+
+                // if empty response is expected, any non related response is dropped
+                if (typeof(TResponse) == typeof(EmptyResponse))
+                {
+                    // if not empty content, check what is there for debug purposes.
+                    if (result.Content.GetType().Name != "EmptyContent")
+                    {
+                        _logger?.LogDebug("Response is not json, but {content}",
```

Comment:
I'm not sure about the log here: shouldn't it be something like "Response is not empty" ?